src/auth/ui/vue/show.ts

```diff
@@ -797,11 +726,77 @@ export async function showAuthWebview(
     if (!subscriptions) {
         subscriptions = [
             webview.onDidDispose(() => {
-                activePanel?.server.emitWebviewClosed()
+                if (activePanel) {
+                    emitWebviewClosed(activePanel.server)
+                }
                 vscode.Disposable.from(...(subscriptions ?? [])).dispose()
                 activePanel = undefined
                 subscriptions = undefined
             }),
         ]
     }
 }
+
+/**
+ * The metric emitted when the webview is closed by the user.
+ */
+export async function emitWebviewClosed(authWebview: ClassToInterfaceType<AuthWebview>) {
+    const [prevFeatureType, prevAuthType] = [authWebview.getPreviousFeatureType(), authWebview.getPreviousAuthType()]
+    if (prevFeatureType && prevAuthType) {
+        // We are closing the webview, and have an auth form if they were
+        // interacting but did not complete it.
+        await authWebview.stopAuthFormInteraction(prevFeatureType, prevAuthType)
+    }
+
+    const authsInitial = authWebview.getAuthsInitial()
+    const authsAdded = authWebview.getAuthsAdded()
+
+    const numConnectionsInitial = authsInitial.size
+    const numConnectionsAdded = authsAdded.length
+
+    const source = authWebview.getSource()
+    const result: Result = determineResult(source, numConnectionsInitial, numConnectionsAdded)
+
+    telemetry.auth_addedConnections.emit({
+        source: source ?? '',
+        result,
+        attempts: authWebview.getTotalAuthAttempts(),
+        reason: 'closedWebview',
+        authConnectionsCount: numConnectionsInitial + numConnectionsAdded,
+        newAuthConnectionsCount: numConnectionsAdded,
+        enabledAuthConnections: buildCommaDelimitedString(new Set([...authsInitial, ...authsAdded])),
+        newEnabledAuthConnections: buildCommaDelimitedString(authsAdded),
+    })
+
+    function determineResult(
+        source: AuthSource | undefined,
+        numConnectionsInitial: number,
+        numConnectionsAdded: number
+    ): Result {
+        let result: Result
+
+        if (numConnectionsAdded > 0) {
+            result = 'Succeeded'
+        } else if (authWebview.getTotalAuthAttempts() > 0) {
+            // There were no new auth connections added, but attempts were made
+            result = 'Failed'
+        } else {
+            // No new auth connections added, but no attempts were made
+            result = 'Cancelled'
```

Comment:
Can we use the run() pattern instead, and throw a "canceled" error to represent cancel, as usual?